impl/client/src/main/javascript/web/pentaho/ccc/visual/Funnel.js

```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright 2010 - 2019 Hitachi Vantara. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "pentaho/module!_",
+  "pentaho/visual/impl/View",
+  "common-ui/echarts",
+  "./_util",
+  "pentaho/visual/color/util"
+], function (module, BaseView, echarts, util, visualColorUtils) {
+
+  "use strict";
+
+  return BaseView.extend(module.id, {
```

Comment:
The base class is correct, however, the namespace `pentaho/ccc/visual` is intended to be used only by views based on the CCC library.

Using `pentaho/echart/visual`, is a possibility, but can become unwieldy if we start using several other graphical libraries... Maybe start putting views from external graphical libraries under `pentaho/visual/views/<...echart...>` would be a good idea...